core/pkg/eval/fractional_evaluation.go

```diff
@@ -32,54 +34,49 @@ func (fe *FractionalEvaluator) FractionalEvaluation(values, data interface{}) in
 	return distributeValue(valueToDistribute, feDistributions)
 }
 
-func parseFractionalEvaluationData(values, data interface{}) (string, []fractionalEvaluationDistribution, error) {
-	valuesArray, ok := values.([]interface{})
+func parseFractionalEvaluationData(values, data any) (string, []fractionalEvaluationDistribution, error) {
+	valuesArray, ok := values.([]any)
 	if !ok {
 		return "", nil, errors.New("fractional evaluation data is not an array")
 	}
 	if len(valuesArray) < 2 {
 		return "", nil, errors.New("fractional evaluation data has length under 2")
 	}
 
-	bucketBy, ok := valuesArray[0].(string)
-	if !ok {
-		return "", nil, errors.New("first element of fractional evaluation data isn't of type string")
-	}
-
-	dataMap, ok := data.(map[string]interface{})
+	dataMap, ok := data.(map[string]any)
 	if !ok {
-		return "", nil, errors.New("data isn't of type map[string]interface{}")
+		return "", nil, errors.New("data isn't of type map[string]any")
 	}
 
-	// Ignore the ok as we can't really do anything if the properties are
+	// Ignore the error as we can't really do anything if the properties are
 	// somehow missing.
 	properties, _ := getFlagdProperties(dataMap)
 
-	v, ok := dataMap[bucketBy]
-	if !ok {
-		return "", nil, nil
-	}
-
-	valueToDistribute, ok := v.(string)
-	if !ok {
-		return "", nil, fmt.Errorf("var: %v isn't of type string", v)
+	bucketBy, ok := valuesArray[0].(string)
+	if ok {
+		valuesArray = valuesArray[1:]
+	} else {
+		bucketBy, ok = dataMap[targetingKeyKey].(string)
+		if !ok {
+			return "", nil, errors.New("targeting key not supplied nor in context")
+		}
 	}
 
 	feDistributions, err := parseFractionalEvaluationDistributions(valuesArray)
 	if err != nil {
 		return "", nil, err
 	}
 
-	return fmt.Sprintf("%s$%s", properties.FlagKey, valueToDistribute), feDistributions, nil
+	return fmt.Sprintf("%s$%s", properties.FlagKey, bucketBy), feDistributions, nil
```

Comment:
```suggestion
	return fmt.Sprintf("%s%s", properties.FlagKey, bucketBy), feDistributions, nil
```
I think this token technically violates the specification. We can actually add it, but I'm dubious of the value, and in other implementations that actually inject the concatenation expression, it might just make the impl more difficult.